plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/HudiAvroSerializer.java

```diff
@@ -199,7 +211,7 @@ else if (type instanceof DecimalType decimalType) {
                     type.writeLong(output, encodeShortScaledValue(decimal, decimalType.getScale()));
                 }
                 else if (type.equals(DATE)) {
-                    type.writeLong(output, ((Number) value).intValue());
+                    type.writeLong(output, ((SqlDate) value).getDays());
```

Comment:
probably need to rebase with hudi-1.0.x as this fix already landed in #5

Reply:
This change is a cherrpick of #7.

Reply:
I will exclude this change for now.